src/javascript/eslint.ts

```diff
@@ -62,6 +61,15 @@ export interface EslintOptions {
    */
   readonly prettier?: boolean;
 
+  /**
+   * The extends array in eslint is order dependent.
+   * Setting this option true will make a best effort to properly sort the extends array
+   * according to ESLint best practices.
+   *
+   * @default true
+   */
+  readonly sortExtends?: boolean;
```

Comment:
I wonder if we should just allow a compare function here:
```suggestion
  readonly sortExtends?: ICompareString;
```

```ts
interface ICompareString {
  compare(a: string, b: string): number;
}
```

And our Eslint component can provide a default:

class Eslint {
  static public SORT_EXTENDS_DEFAULT(): ICompareString {
    return {
      compare(a: string, b: string): number {
          /// to do
      }
    };
  }
}

Reply:
I like this idea.  I assume the nesting is a JSII requirement, as I see the CDK do that too, and I can't imagine .NET or java having a function parameter...

Reply:
Yes, unfortunately free-floating functions are not jsii compatible. So we need a protocol interface for it.